### ISSUE TITLE
[Bug] Delay before TablePlus appears when switching from other applications

### ISSUE THREAD

**Issue by johanrosenson** [TablePlus/TablePlus#1888]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0): 
MySQL 5.7

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
3.5.0 (308)
MacOS Catalina 10.15.4

3. The steps to reproduce this issue: 
Don't have specific steps, it just happens.

Noted: If the bug is related to data, please attach an example SQL data.
- I do not think this is related to data, found no correlation based on what i am doing inside TablePlus.

Description:
When switching (mostly using CMD+Tab) to TablePlus from other applications i notice a slight delay before TablePlus get focus, this makes the workflow slow since I have to wait for TablePlus.

This behaviour also makes me second guess if i actually did switch to TablePlus or if i failed pressing CMD+Tab, so i do it again, but this second time the CMD+Tab will just switch back to the previous application making me have to start the switching all over again, it gets very confusing fast.

While the delay most of the time is short its enough to be noticeable and halt my workflow, and sometimes the delay can be up to 3-5 seconds.

I'm attaching two screen recordings that will show the delay, in the first recording I am using CMD+Tab, and in the second recording I am using the mouse (there it can clearly be seen that there is a delay from my click and to when TablePlus gets focus).
It seems like the Application TablePlus becomes active instantly, but the delay is when TablePlus is putting focus on the actual Window.

I have not experienced this with any other of my applications so i am almost 100% certain this is an TablePlus issue, some of the other tools im using right now is Sublime Text, GitKraken, Insomnia and Slack, and all of them get instant focus when switching, just to rule out that my system is slow in general.

Thanks! :) 

Switching using Cmd+Tab:
![TablePlus-cmd-tab-optimized](https://user-images.githubusercontent.com/10432296/82141746-2567d480-9862-11ea-98ad-81732e45718b.gif)

Switching using mouse click:
![TablePlus-click-optimized](https://user-images.githubusercontent.com/10432296/82141749-28fb5b80-9862-11ea-9774-08ab4166202c.gif)

**Comment by johanrosenson** [comment]
Just saw this issue that appears to be the same issue:

https://github.com/TablePlus/TablePlus/issues/1877

**Comment by huyphams** [comment]
Hey thank you for letting me know @johanrosenson please try this patch: https://www.dropbox.com/s/am0f0toy4wjf9pl/TablePlus.dmg?dl=0

**Comment by johanrosenson** [comment]
@huyphams I think i was too slow, the patch is not available any more.

**Comment by huyphams** [comment]
Could you please try this link @johanrosenson https://www.dropbox.com/s/ikxmp2miipd5jql/tableplus.dmg?dl=0 
Thanks!

**Comment by johanrosenson** [comment]
@huyphams My first impressions is that this resolved it, can't be 100% sure yet since i have not used it much the last two days, and the delay was intermittent and would not always be present.

But, it's looking good so far, i will reply again if i see the problem returning.

**Comment by huyphams** [comment]
It would be slowed down if you opened a table with a lot of text columns, and you selected that tab before switching or opened a right sidebar with a big text.

**Comment by johanrosenson** [comment]
@huyphams I can rule that out with 100% certainty, i have only been working with smaller datasets recently and nothing that could explain a slowdown.

Sometimes the slowdown came directly after connecting to a database, nothing open, no queries executed.

The positive news is that i think your fix helped, so far it has been better than before. what did you change?

**Comment by huyphams** [comment]
I've changed a lot, moved some highlight syntax code to the background thread. Please let me know if you're still seeing this issue.

**Comment by andreaolivato** [comment]
Hi! 

I was experiencing the same exact delay as @johanrosenson .

The patch you attached 6 days ago completely solved the issue.

Many thanks and let me know if you need any more testing to zeroing on the problem

**Comment by huyphams** [comment]
I'm going to release an update because of this issue, thanks @andreaolivato @johanrosenson!

**Comment by johanrosenson** [comment]
@huyphams i must report that i feel the issue have returned, i am running build 320 now, the delay is back but not as much as before, but still noticeable.

Do you experience the same @andreaolivato ?

**Comment by andreaolivato** [comment]
@johanrosenson I am on build 320 as well, not experiencing any delays at the moment.

**Comment by cjke** [comment]
I'm on 322 and seeing similar behavior. 

Dataset ~20 columns (mixed varchar, int, text) x ~50 rows on a mysql connection yields about a ~1 sec delay.

It happens when alt-tabbing or clicking the dock icon. 
When clicking the window directly from the background (like the original poster above) then it is fine. 

Hardware: 2016 Macbook Pro, i7 16GB

**Comment by johanrosenson** [comment]
@huyphams this issue appears to be back, running build 336 and noticing delays again in the interface, same as before, if i would create a new issue i could include the exact same gif's as in the start of this issue :)

**Comment by jcasanova** [comment]
Hi !

TLDR; close too many opened connections.

![image](https://user-images.githubusercontent.com/1027644/148449045-c5839c10-380f-48b7-988f-ad5a626dfb49.png)

Found this issue while encountering the same behavior. Was ready to ditch TablePlus for this lag (3 secs are very annoying) 😱 
Then I realized over the time I was keping all my connections in the same window (~8/10 connections).
Switched back to 1 or 2 connections in the same window and pb is over.

Hope this helps.
